### PR TITLE
Add `element_types` kwarg to example notebook

### LINKED ISSUE #609: [Bug]: Not able to reproduce validation MAE using matgl
### Email (Optional)

as7959@princeton.edu

### Version

v1.2.7

### Which OS(es) are you using?

- [ ] MacOS
- [ ] Windows
- [x] Linux

### What happened?

I'm a student in @Andrew-S-Rosen's Group

**Problem**
During the training of a MEGNet model, we are getting a validation MAE that is different than the validation MAE that we get when we apply the trained model on the validation set outside of the training process. I replicate the example notebook "[Training a MEGNet Formation Energy Model with PyTorch Lightning.ipynb](https://github.com/materialsvirtuallab/matgl/blob/main/examples/Training%20a%20MEGNet%20Formation%20Energy%20Model%20with%20PyTorch%20Lightning.ipynb)"

Versions:
matgl         1.2.7
lightning:    2.5.1
Python:       3.10.18

### Code snippet
[data.json](https://github.com/user-attachments/files/20928360/data.json)

```Python import warnings

import lightning as L
import torch
from dgl.data.utils import split_dataset
from lightning.pytorch.loggers import CSVLogger
from matgl.ext.pymatgen import Structure2Graph, get_element_list
from matgl.graph.data import MGLDataLoader, MGLDataset, collate_fn_graph
from matgl.layers import BondExpansion
from matgl.models import MEGNet
from matgl.utils.training import ModelLightningModule
from monty.serialization import loadfn
from pymatgen.core import Structure
from sklearn.metrics import mean_absolute_error
import warnings

warnings.simplefilter("ignore")

# Load dataset
data = loadfn("data.json")
structures = []
mp_ids = []
eform_per_atom = []
for entry in data:
    structures.append(Structure.from_str(entry["structure"], fmt="cif"))
    mp_ids.append(entry["material_id"])
    eform_per_atom.append(entry["formation_energy_per_atom"])

# Make model
elem_list = get_element_list(structures)
converter = Structure2Graph(element_types=elem_list, cutoff=4.0)
mp_dataset = MGLDataset(
    structures=structures,
    labels={"Eform": eform_per_atom},
    converter=converter,
)

train_data, val_data, test_data = split_dataset(
    mp_dataset,
    frac_list=[0.8, 0.1, 0.1],
    shuffle=False,
    random_state=42,
)
train_loader, val_loader, test_loader = MGLDataLoader(
    train_data=train_data,
    val_data=val_data,
    test_data=test_data,
    collate_fn=collate_fn_graph,
    batch_size=2,
    num_workers=0,
)

node_embed = torch.nn.Embedding(len(elem_list), 16)
bond_expansion = BondExpansion(
    rbf_type="Gaussian", initial=0.0, final=5.0, num_centers=100, width=0.5
)

model = MEGNet(
    dim_node_embedding=16,
    dim_edge_embedding=100,
    dim_state_embedding=2,
    nblocks=3,
    hidden_layer_sizes_input=(64, 32),
    hidden_layer_sizes_conv=(64, 64, 32),
    nlayers_set2set=1,
    niters_set2set=2,
    hidden_layer_sizes_output=(32, 16),
    is_classification=False,
    activation_type="softplus2",
    bond_expansion=bond_expansion,
    cutoff=4.0,
    gauss_width=0.5,
)

lit_module = ModelLightningModule(model=model)

logger = CSVLogger("logs", name="MEGNet_training")
trainer = L.Trainer(max_epochs=20, logger=logger)
trainer.fit(
    model=lit_module, train_dataloaders=train_loader, val_dataloaders=val_loader
)

val_structures = [structures[i] for i in val_data.indices]
val_eform_per_atom = [eform_per_atom[i] for i in val_data.indices]

output = [model.predict_structure(structure).item() for structure in val_structures]
mae = mean_absolute_error(val_eform_per_atom, output)

print(f"calculated manual MAE {mae}")

```

### Log output

```shell
trainer.fit output ->
GPU available: True (cuda), used: True
  | Name  | Type              | Params | Mode
----------------------------------------------------
0 | model | MEGNet            | 189 K  | train
1 | mae   | MeanAbsoluteError | 0      | train
2 | rmse  | MeanSquaredError  | 0      | train
----------------------------------------------------
189 K     Trainable params
100       Non-trainable params
189 K     Total params
0.758     Total estimated model params size (MB)
109       Modules in train mode
0         Modules in eval mode
Epoch 19: 100%|█| 40/40 [00:01<00:00, 33.77it/s, v_num=10, val_Total_Loss=0.290, val_MAE=0.402, val_RMSE=0.476,`Trainer.fit` stopped: `max_epochs=20` reached.
Epoch 19: 100%|█| 40/40 [00:01<00:00, 33.13it/s, v_num=10, val_Total_Loss=0.290, val_MAE=0.402, val_RMSE=0.476,
calculated manual MAE 1.4439821497793355
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Andrew-S-Rosen**
## Summary

This PR updates the "[Training a MEGNet Formation Energy Model with PyTorch Lightning.ipynb](https://github.com/materialsvirtuallab/matgl/blob/main/examples/Training%20a%20MEGNet%20Formation%20Energy%20Model%20with%20PyTorch%20Lightning.ipynb)" to add the `element_types` keyword argument to the `MEGNet()` class. Without this, the user is likely to make an error when making predictions with the trained model.

Closes #609.

### CODE REVIEW

**Comment by Andrew-S-Rosen** [comment]
@shyuep: FYI, the linting errors are seemingly unrelated to this PR.
